models/User.php

```diff
@@ -69,16 +71,87 @@ public function behaviors()
     public function rules()
     {
         return [
-            ['status', 'default', 'value' => self::STATUS_ACTIVE],
-            ['status', 'in', 'range' => [self::STATUS_ACTIVE, self::STATUS_DELETED]],
             ['is_authenticated', 'boolean'],
-            ['name', 'string'],
             [['gender_id', 'sexuality_id', 'currency_id'], 'integer'],
+
+            ['status',
+                'default',
+                'value' => self::STATUS_ACTIVE],
+            ['status',
+                'in',
+                'range' => [self::STATUS_ACTIVE, self::STATUS_DELETED]],
+
             ['email', 'email'],
+            ['email', 'required'],
+            ['email',
+                'unique',
+                'message' => 'Email must be unique.'
+            ],
+
+            ['username', 'required'],
+            ['username', 'trim'],
+            ['username',
+                'match',
+                'pattern' => '/^[a-zA-Z0-9_]+$/i',
+                'message' => 'Username must contain letters, numbers and _'
+            ],
+            ['username', 'validateUsernameUnique'],
+
+            ['name', 'string'],
+            ['name', 'required'],
+            ['name', 'trim'],
+            ['name', 'validateNameString'],
+
+            ['birthday', 'required'],
+            ['birthday', 'date'],
+
             [['timezone'], 'default', 'value' => 'UTC'],
+            ['timezone', 'required'],
+
+            ['gender', 'required'],
+
+            ['currency', 'required'],
+
+            ['sexuality', 'required'],
```

Comment:
```suggestion
            [ [ 'timezone', 'gender', 'currency', 'sexuality' ], 'required' ],
```
But you must specify such validators for real attributes: for example, User model doesn't contain 'currency' attribute - it contains 'currency_id' field, so 'currency_id' is required, not 'currency'.